### PR TITLE
Use lowdash's sluggify method to ensure the appname is valid.

### PR DESCRIPTION
**Pull request by heavysixer**
This is a fix for issue: #25

### CODE REVIEW

**Comment by SBoudrias** [comment]
Can you add a test?

**Comment by heavysixer** [comment]
sure thing, let me do that now, also I guess I need to add lowdash as a dependency because i thought it was getting injected by the yeoman generator because I saw it used _package.json but apparently it's not.

**Comment by heavysixer** [comment]
@SBoudrias is there documentation on how to run these existing tests? I don't see any in the repo and its non-obvious from looking at the source.

**Comment by SBoudrias** [comment]
run `npm test`

**Comment by heavysixer** [comment]
Hey @SBoudrias, I've added a test and added lowdash to the package dependencies. Please let me know if you need anything else.

**Comment by heavysixer** [comment]
I see that the Travis CI build is failing on io.js however I have no idea why it is getting this error: 

`Uncaught Error: Unable to write "package.json" file (Error code: EACCES).`

This is not something i believe that I modified.

**Comment by SBoudrias** [comment]
Can you use the latest lodash version?

**Comment by heavysixer** [comment]
sure i bound it to the one that was used in yeoman generator 
didn't know if it was done that way on purpose or not.

**Comment by heavysixer** [comment]
@SBoudrias I believe i've squashed my commits together. Let me know if I did something incorrectly, it's the first time i've tried to do that.

**Comment by SBoudrias** [comment]
Looks good, but when squashing correctly, there should be a single commit in this PR.

**Comment by SBoudrias** [comment]
Also, check the tests, they're all failing at the moment.

**Comment by heavysixer** [comment]
@SBoudrias please see my note about the failing test above.

**Comment by SBoudrias** [comment]
Yeah, it's not that error currently.

**Comment by heavysixer** [comment]
hmmm maybe my squash messed something up. I feel like i'm failing open source. Thanks for your help.

**Comment by heavysixer** [comment]
the tests all pass fine for me locally. I think i am sort of stuck atm.

**Comment by SBoudrias** [comment]
Delete your node_modules folder and run `npm install`, chances are it's not in the same state as a new fresh one.

**Comment by heavysixer** [comment]
yes you are right it appears that the new version of lowdash does not have slugify However, your code presently uses that function elsewhere. Should I revert back to the previous version of lowdash?

**Comment by SBoudrias** [comment]
yeoman used `underscore.string`, but I'm pretty sure lodash 3 have strings methods. Go check the documentation, the name might be different.

**Comment by heavysixer** [comment]
@SBoudrias specifically i am talking about this https://github.com/yeoman/generator-gruntfile/blob/master/app/templates/_package.json#L2

**Comment by SBoudrias** [comment]
Yeah, that is underscore.string, and it's a deprecated feature. Either require `underscore.string` or find a matching method in lodash.

**Comment by heavysixer** [comment]
yeah but won't the tests continue to fail without that file also being changed or are you saying it's not presently being used?

**Comment by SBoudrias** [comment]
It'll work fine.

**Comment by heavysixer** [comment]
@SBoudrias ok awesome looks like everything passes again, thanks for taking the time to help me out.

File: package.json [hunk 1]
```diff
@@ -28,6 +28,8 @@
   },
   "dependencies": {
     "grunt": "~0.4.1",
+    "lodash": "^2.4.1",
+    "underscore.string": "^3.0.3",
```
Comment (SBoudrias) [inline package.json]: you don't need underscore.string if you don't require it.

Comment (heavysixer) [inline package.json]: the problem i am having running tests is that unless i include underscore.string or lodash and cast it to the 
"_" variable then we end up with an undefined value in the code. However, there is no reason to have both in there. I am moving too fast trying to get this PR right. My apologies for that.

File: test/test.js [hunk 2]
```diff
@@ -85,6 +85,20 @@ describe('gruntfile:app with existing package.json', function () {
     });
   });
 
+  it('should slugify the appname properly', function(done){
```
Comment (arthurvr) [inline test/test.js]: Code style

```js
it('should slugify the appname properly', function (done) {
```

**Comment by SBoudrias** [comment]
Good to merge. Thanks for taking care of it.

**Pull request merged**